python_modules/dagster/dagster/core/utils.py

```diff
@@ -80,3 +81,14 @@ def check_dagster_package_version(library_name, library_version):
             __version__, library_name, library_version
         )
         warnings.warn(message)
+
+
+def parse_env_var(env_var_str: str) -> Tuple[str, str]:
+    if "=" in env_var_str:
+        split = env_var_str.split("=")
```

Comment:
This will cause issues if you have `=` in your environment variable (very likely for anything base64 encoded). Something like:

```suggestion
        split = env_var_str.split("=", maxsplit=1)
```

will probably be sufficient because I don't think you can have `=` in the name of an envvar, but we should add a few unit tests of this function to assert against various edge cases we can imagine (empty list, list with only keys, list with names and key/value pairs, values with an equal sign, values with multiple equal signs, etc.)